Source/Charts/Data/Implementations/Standard/ChartDataEntry.swift

```diff
@@ -108,32 +99,18 @@ open class ChartDataEntry: ChartDataEntryBase
     }
 }
 
-public func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
-{
-    if lhs === rhs
-    {
-        return true
-    }
-    
-    if !lhs.isKind(of: type(of: rhs))
+// MARK: Equatable
+extension ChartDataEntry/*: Equatable*/ {
+    public static func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
     {
-        return false
-    }
-    
-    if lhs.data !== rhs.data && !lhs.data!.isEqual(rhs.data)
-    {
-        return false
-    }
-    
-    if fabs(lhs.x - rhs.x) > Double.ulpOfOne
-    {
-        return false
-    }
-    
-    if fabs(lhs.y - rhs.y) > Double.ulpOfOne
-    {
-        return false
+        if lhs === rhs
+        {
+            return true
+        }
+
+        return lhs.isKind(of: type(of: rhs))
+            && (lhs.data?.isEqual(rhs.data) ?? true)
```

Comment:
>!== shouldn't be used

Why?
The old code wanted to check the data identity first (`lhs.data !== rhs.data`) and later the data equality. However new code just checks `(lhs.data == nil && rhs.data == nil)`, which is different I think. If someone put same data in LineChartData and BarChartData, the new code shall return  true, while it is supposed to be false?

BTW, if you mean you want to turn `ChartDatEntry` into struct and make a PR, we are grateful and thank for your effort, but the main author has been missing for a while, I am not sure if all your work could get merged in time. So take your time and don't let this occupied most of your time and you find it's not getting attention will make you kind of frustrated.